File: AVeryBigSum.c
/*
You are given an array of integers of size . You need to print the sum of the elements in the array, keeping in mind that some of those integers may be quite large.

Input Format

The first line of the input consists of an integer . The next line contains  space-separated integers contained in the array.

Output Format

Print a single value equal to the sum of the elements in the array.
*/

#include <math.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <assert.h>
#include <limits.h>
#include <stdbool.h>

long int aVeryBigSum(int n, int ar_size, long int* ar) 
{
    // Complete this function
    long int sum = 0;
    for(int count = 0; count < ar_size; count++)
    {
        sum += ar[count];
    }
    return sum;
}

int main() {
    int n; 
    scanf("%i", &n);
    long int *ar = malloc(sizeof(long int) * n);
    for(int ar_i = 0; ar_i < n; ar_i++){
       scanf("%li",&ar[ar_i]);
    }
    long int result = aVeryBigSum(n, n, ar);
    printf("%ld\n", result);
    return 0;
}

File: AVeryBigSum.py
#!/bin/python

import sys

def aVeryBigSum(n, ar):
    # Complete this function
    #5
    #1000000001 1000000002 1000000003 1000000004 1000000005
    sum = 0
    for count in range(n):
        sum += ar[count]
    return sum

n = int(raw_input().strip())
ar = map(long, raw_input().strip().split(' '))
result = aVeryBigSum(n, ar)
print(result)
